### DISCUSSION TITLE
No switching from absorption to float

### DISCUSSION THREAD

**Discussion by regda1986** [Louisvdw/dbus-serialbattery#1011] [Q&A]
Hey guys,

I have a small problem with my serial battery settings. The adapter does not switch to float mode. Today its stays in absorption mode twice for over 1.5 hours without going into float. Maximum cell difference was less than 0.005V in the second time window.

I noticed that the MP2 had an output voltage of 55.18 to 55.22V (thats fits). But the JK BMS only reported 55.05 to 55.10V. Could that be the problem? Charging current was only 1-2A, so the voltage drop should be very low with 35mm² and only 1.5m cable length.

Setup: MP2 3000, Venus OS on RPi4, 16x 160Ah EVE cells with JK-BMS B2A24S15P

Greetings and thanks,
Martin

My config.ini, rest is stock:

`[DEFAULT]

; If you want to add custom values/settings, then check the values/settings you want to change in "config.default.ini"
; and insert them below to persist future driver updates.

; Example (remove the semicolon ";" to uncomment and activate the value/setting):
MAX_BATTERY_CHARGE_CURRENT = 50.0
MAX_BATTERY_DISCHARGE_CURRENT = 70.0

MIN_CELL_VOLTAGE   = 3.000
; Max voltage can seen as absorption voltage
MAX_CELL_VOLTAGE   = 3.450
FLOAT_CELL_VOLTAGE = 3.375

; --------- Cell Voltage Current limitation (affecting CCL/DCL) ---------
; Description: Maximal charge / discharge current will be in-/decreased depending on min and max cell voltages
; Example: 18 cells * 3.55V/cell = 63.9V max charge voltage
;          18 cells * 2.70V/cell = 48.6V min discharge voltage
;          But in reality not all cells reach the same voltage at the same time. The (dis)charge current
;          will be (in-/)decreased, if even ONE SINGLE BATTERY CELL reaches the limits

; Charge current control management referring to cell-voltage enable (True/False).
CCCM_CV_ENABLE = True
; Discharge current control management referring to cell-voltage enable (True/False).
DCCM_CV_ENABLE = True

; Set steps to reduce battery current
; The current will be changed linear between those steps if LINEAR_LIMITATION_ENABLE is set to True

CELL_VOLTAGES_WHILE_CHARGING = 3.55, 3.50, 3.44, 3.30
MAX_CHARGE_CURRENT_CV_FRACTION = 0, 0.05, 0.35, 1

CELL_VOLTAGES_WHILE_DISCHARGING   = 2.70, 2.80, 2.90, 3.10
MAX_DISCHARGE_CURRENT_CV_FRACTION = 0, 0.1, 0.5, 1

; --------- SOC limitation (affecting CCL/DCL) ---------
; Description: Maximal charge / discharge current will be increased / decreased depending on State of Charge,
;              see CC_SOC_LIMIT1 etc.
; Example: The SoC limit will be monitored to control the currents.
; Charge current control management enable (True/False).
CCCM_SOC_ENABLE = False
; Discharge current control management enable (True/False).
DCCM_SOC_ENABLE = False`

**Comment by mdkeil** [comment]
I think the serialbattery-driver never reached MAX_CELL_VOLTAGE * 16;  You can try to use VOLTAGE_DROP.

```
; --------- Voltage drop ---------
; If you have a voltage drop between the BMS and the charger because of wire size or length
; then you can specify the voltage drop here. The driver will then add the voltage drop
; to the calculated CVL to compensate.
; Example:
;     cell count: 16
;     MAX_CELL_VOLTAGE = 3.45
;     max voltage calculated = 16 * 3.45 = 55.20
;     CVL is set to 55.20 V and the battery is now charged until the charger reaches 55.20 V.
;     The BMS now measures 55.05 V since there is a voltage drop of 0.15 V on the cable.
;     Since the dbus-serialbattery reads the voltage of 55.05 V from the BMS the max voltage
;     of 55.20 V is never reached and max voltage is kept forever.
;     By setting the VOLTAGE_DROP to 0.15 V the voltage on the charger is increased and the
;     target voltage on the BMS is reached.
VOLTAGE_DROP = 0.00
```

**Reply by regda1986** [reply]
Many thanks! Thats was the solution for my problem. :)
